backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -751,6 +755,28 @@ public FutureExperimentRunDAO(
                                             }),
                                     executor);
 
+                            final var futureCodeVersionFromBlob =
+                                repoIdsFutureTasks.thenCompose(
+                                    selfAllowedRepositoryIds ->
+                                        codeVersionFromBlobHandler.getExperimentRunCodeVersionMap(
+                                            ids, selfAllowedRepositoryIds),
+                                    executor);
+                            futureBuildersStream =
+                                futureBuildersStream.thenCombine(
+                                    futureCodeVersionFromBlob,
+                                    (stream, runCodeVersionConfigBlob) ->
+                                        stream.map(
+                                            builder -> {
+                                              if (!runCodeVersionConfigBlob.isEmpty()
+                                                  && runCodeVersionConfigBlob.containsKey(
+                                                      builder.getId())) {
+                                                builder.putAllCodeVersionFromBlob(
+                                                    runCodeVersionConfigBlob.get(builder.getId()));
+                                              }
+                                              return builder;
+                                            }),
+                                    executor);
```

Comment:
The more I read this code pattern, the more I get a sense that it shouldn't be here polluting the main method but instead hidden in the handler. We can do as a follow up, but this method is _gigantic_ already and really hard for a reader to follow